src/migrations/20181109151249_transactionHashes.ts

```diff
@@ -0,0 +1,15 @@
+import * as Knex from "knex";
+
+exports.up = async (knex: Knex): Promise<any> => {
+  return knex.schema.dropTableIfExists("transactionHashes").then((): PromiseLike<any> => {
+    return knex.schema.raw(`CREATE TABLE transactionHashes (
+      "transactionHash" varchar(66) PRIMARY KEY NOT NULL,
```

Comment:
This shouldn't be primary key. on removal, the removed flag is set, but the same transaction is likely to come back, creating a duplicate primary key issue. then when it comes back, we don't have information about when the new tx showed up, so our data is incomplete. I'll remove this, add blockHash, and create a unique index on blockHash/transactionHash.